### ISSUE TITLE
`#colorLiteral` triggers `number_separator`

### ISSUE THREAD

**Issue by awein** [realm/SwiftLint#4576]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`#colorLiteral` will trigger `number_separator` three times starting 0.50.0

E.g.
```
let color = #colorLiteral(red: 0.3543982506, green: 0.318749547, blue: 0.6367015243, alpha: 1)
```

##### Complete output when running SwiftLint, including the stack trace and command used

```
Contents.swift:7:32: warning: Number Separator Violation: Underscores should be used as thousand separator in large decimal numbers. (number_separator)
Contents.swift:7:53: warning: Number Separator Violation: Underscores should be used as thousand separator in large decimal numbers. (number_separator)
Contents.swift:7:72: warning: Number Separator Violation: Underscores should be used as thousand separator in large decimal numbers. (number_separator)
```

### Environment

* SwiftLint version: 0.50.0
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - number_separator
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  * No
* Which Xcode version are you using?
  * Xcode 14.1
* Do you have a sample that shows the issue?
```swift
// This triggers a violation:
let color = #colorLiteral(red: 0.3543982506, green: 0.318749547, blue: 0.6367015243, alpha: 1)
```

**Comment by jpsim** [comment]
The `#colorLiteral` is unrelated to the issue here, the `number_separator` rule's default configuration expects `_` separators in groups of 3 for the fractional component. In other words, the rule expects your line to look like this:

```swift
let color = #colorLiteral(red: 0.354_398_250_6, green: 0.318_749_547, blue: 0.636_701_524_3, alpha: 1)
```

However, I think this is a bug where after the rule rewrite in https://github.com/realm/SwiftLint/pull/4333, the default configuration changed to require `_` separators in the fractional components of float literals.
